### PR TITLE
upgrade to java 21

### PR DESCRIPTION
**Pull request by nhill-ch**

### CODE REVIEW

**Comment by akunati-ch** [comment]
@nhill-ch , Can you update the README.md file with correct version please

File: src/test/java/uk/gov/companieshouse/api/accounts/CompanyAccountsApplicationTest.java [hunk 1]
```diff
@@ -4,10 +4,7 @@
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.TestInstance;
 import org.junit.jupiter.api.extension.ExtendWith;
-import org.mockito.InOrder;
-import org.mockito.InjectMocks;
-import org.mockito.Mock;
-import org.mockito.Mockito;
+import org.mockito.*;
```
Comment (hwatkins-ch) [inline src/test/java/uk/gov/companieshouse/api/accounts/CompanyAccountsApplicationTest.java:7]: Wildcard import

File: src/test/java/uk/gov/companieshouse/api/accounts/controller/CompanyAccountControllerTest.java [hunk 2]
```diff
@@ -1,13 +1,12 @@
 package uk.gov.companieshouse.api.accounts.controller;
 
-import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (hwatkins-ch) [inline src/test/java/uk/gov/companieshouse/api/accounts/controller/CompanyAccountControllerTest.java:3]: Wildcard import

File: src/test/java/uk/gov/companieshouse/api/accounts/service/impl/CompanyAccountServiceImplTest.java [hunk 3]
```diff
@@ -12,12 +12,9 @@
 import static org.mockito.Mockito.when;
 
 import java.io.IOException;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Map;
-import java.util.Optional;
+import java.util.*;
```
Comment (hwatkins-ch) [inline src/test/java/uk/gov/companieshouse/api/accounts/service/impl/CompanyAccountServiceImplTest.java:15]: Wildcard import

File: src/test/java/uk/gov/companieshouse/api/accounts/service/impl/FilingServiceImplTest.java [hunk 4]
```diff
@@ -1,7 +1,6 @@
 package uk.gov.companieshouse.api.accounts.service.impl;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (hwatkins-ch) [inline src/test/java/uk/gov/companieshouse/api/accounts/service/impl/FilingServiceImplTest.java:3]: Wildcard import

File: src/test/java/uk/gov/companieshouse/api/accounts/validation/CreditorsWithinOneYearValidatorTests.java [hunk 5]
```diff
@@ -7,12 +7,9 @@
...
 import static org.mockito.Mockito.when;
 
-import javax.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletRequest;
 
-import org.junit.jupiter.api.BeforeEach;
-import org.junit.jupiter.api.DisplayName;
-import org.junit.jupiter.api.Test;
-import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.*;
```
Comment (hwatkins-ch) [inline src/test/java/uk/gov/companieshouse/api/accounts/validation/CreditorsWithinOneYearValidatorTests.java:12]: Wildcard import

File: src/test/java/uk/gov/companieshouse/api/accounts/validation/LoanValidatorTest.java [hunk 6]
```diff
@@ -21,19 +21,16 @@
...
 
-import javax.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletRequest;
 import java.util.ArrayList;
 import java.util.List;
 
-import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertTrue;
-import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (hwatkins-ch) [inline src/test/java/uk/gov/companieshouse/api/accounts/validation/LoanValidatorTest.java:28]: Wildcard import

File: src/test/java/uk/gov/companieshouse/api/accounts/validation/transactionclosure/PerviousPeriodTxnClosureValidatorTest.java [hunk 7]
```diff
@@ -1,14 +1,12 @@
 package uk.gov.companieshouse.api.accounts.validation.transactionclosure;
 
-import static junit.framework.TestCase.assertEquals;
-import static junit.framework.TestCase.assertTrue;
-import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (hwatkins-ch) [inline src/test/java/uk/gov/companieshouse/api/accounts/validation/transactionclosure/PerviousPeriodTxnClosureValidatorTest.java:3]: Wildcard import

File: src/test/java/uk/gov/companieshouse/api/accounts/validation/transactionclosure/StocksTxnClosureValidatorTest.java [hunk 8]
```diff
@@ -28,19 +28,16 @@
...
 
-import javax.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletRequest;
 import java.util.HashMap;
 import java.util.Map;
 
-import static junit.framework.TestCase.assertEquals;
-import static junit.framework.TestCase.assertFalse;
-import static junit.framework.TestCase.assertTrue;
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (hwatkins-ch) [inline src/test/java/uk/gov/companieshouse/api/accounts/validation/transactionclosure/StocksTxnClosureValidatorTest.java:35]: Wildcard import

**Review by hwatkins-ch — CHANGES_REQUESTED** [review]
Looks good, would like to avoid PR's this big in general though going forward :) 

Few comments around wildcard imports.

File: src/test/java/uk/gov/companieshouse/api/accounts/CompanyAccountsApplicationTest.java (hunk 1 above)
Comment (nhill-ch) [reply src/test/java/uk/gov/companieshouse/api/accounts/CompanyAccountsApplicationTest.java]: done

File: src/test/java/uk/gov/companieshouse/api/accounts/controller/CompanyAccountControllerTest.java (hunk 2 above)
Comment (nhill-ch) [reply src/test/java/uk/gov/companieshouse/api/accounts/controller/CompanyAccountControllerTest.java]: done

File: src/test/java/uk/gov/companieshouse/api/accounts/service/impl/CompanyAccountServiceImplTest.java (hunk 3 above)
Comment (nhill-ch) [reply src/test/java/uk/gov/companieshouse/api/accounts/service/impl/CompanyAccountServiceImplTest.java]: done

File: src/test/java/uk/gov/companieshouse/api/accounts/service/impl/FilingServiceImplTest.java (hunk 4 above)
Comment (nhill-ch) [reply src/test/java/uk/gov/companieshouse/api/accounts/service/impl/FilingServiceImplTest.java]: done

File: src/test/java/uk/gov/companieshouse/api/accounts/validation/CreditorsWithinOneYearValidatorTests.java (hunk 5 above)
Comment (nhill-ch) [reply src/test/java/uk/gov/companieshouse/api/accounts/validation/CreditorsWithinOneYearValidatorTests.java]: done

File: src/test/java/uk/gov/companieshouse/api/accounts/validation/LoanValidatorTest.java (hunk 6 above)
Comment (nhill-ch) [reply src/test/java/uk/gov/companieshouse/api/accounts/validation/LoanValidatorTest.java]: done

File: src/test/java/uk/gov/companieshouse/api/accounts/validation/transactionclosure/PerviousPeriodTxnClosureValidatorTest.java (hunk 7 above)
Comment (nhill-ch) [reply src/test/java/uk/gov/companieshouse/api/accounts/validation/transactionclosure/PerviousPeriodTxnClosureValidatorTest.java]: done

**Review by hwatkins-ch — APPROVED** [review]

**Review by akunati-ch — APPROVED** [review]

**Pull request merged**